### ISSUE TITLE
build.ninja not generating correct command line when path to (eg. clang) has a space in it (eg. "Program Files")

### ISSUE THREAD

**Issue by Kh4n** [mesonbuild/meson#8780]
**Describe the bug**
The `build.ninja` file generated has the spaces in the path removed:
```
rule cpp_COMPILER
 command = "C:\Program" "Files\LLVM\bin\clang++.exe" $ARGS -MD -MQ $out -MF $DEPFILE -o $out "-c" $in
                       ^ if you change this to just a space instead of (" ") it works
```
This results in this error:
```
[1/2] Compiling C++ object ex.exe.p/main.cpp.obj
FAILED: ex.exe.p/main.cpp.obj
"C:\Program" "Files\LLVM\bin\clang++.exe" "-Iex.exe.p" "-I." "-I.." "-fcolor-diagnostics" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-Wnon-virtual-dtor" 
"-g" -MD -MQ ex.exe.p/main.cpp.obj -MF "ex.exe.p\main.cpp.obj.d" -o ex.exe.p/main.cpp.obj "-c" ../main.cpp
CreateProcess failed: The system cannot find the file specified.
ninja: build stopped: subcommand failed.
```
The spaces are formatted incorrectly in the `compile_commands.json` as well.

**To Reproduce**
`meson.build`:
```
project( 'example', 'cpp', version: '1')
executable('ex', 'main.cpp')
```
`main.cpp`:
```
#include <iostream>
int main() { std::cout << "Hello world!" << std::endl; }
```

**Expected behavior**
It should compile.

**system parameters**
* Not a cross build
* Windows 10
* Python 3.9.4
* Meson 0.58.0
* Ninja 1.10.2

**Comment by Kh4n** [comment]
I think I have tracked down the issue to https://github.com/mesonbuild/meson/blob/master/mesonbuild/environment.py#L881l where the `mesonlib.split_args` function is called. This splits the string incorrectly, meaning the real issue is probably https://github.com/mesonbuild/meson/blob/master/mesonbuild/mesonlib/universal.py#L967

**Comment by xclaesse** [comment]
You set `CC=C:\Program Files\LLVM\bin\clang++.exe` in your env?

**Comment by Kh4n** [comment]
Yes, perhaps I should quote this variable?

**Comment by xclaesse** [comment]
yes you have to quote it because it could be a list of args, like `CC=ccache /path to your/gcc`.

**Comment by Kh4n** [comment]
That's fair. Do you think it's worth adding flexibility for new users? Or perhaps an error message? I am willing to add this considering how confusing this error message was for me 😃

**Comment by bonzini** [comment]
This should have been fixed in eb60c041f9d2ae8554045b50080bbfc7b439af01. The original test case was user error, the second one from @mundak needs retesting.
